driver/ixsystems/common.py

```diff
@@ -600,3 +602,92 @@ def create_export(self, volume_name):
             f'{freenas_volume["target"]} {freenas_volume["iqn"]}'
         LOG.debug(f'provider_location: {handle}')
         return handle
+
+    def replicate_volume_from_snapshot(self, target_volume_name, snapshot_name, src_volume_name):
+        """Use replicate api (zfs send/recv) to create a volume from snapshot """
+        source_name = self.configuration.ixsystems_dataset_path + "/" + src_volume_name
+        target_name = self.configuration.ixsystems_dataset_path + "/" + target_volume_name
+        split_snapshot_name = snapshot_name.split("-")
+        replication_name = f'Create volume {target_volume_name} from {src_volume_name}@{split_snapshot_name[0]}-{split_snapshot_name[1]}'
+        naming_schema = f'{split_snapshot_name[0]}-{split_snapshot_name[1]}-%Y-%m-%d-%H-%M'
+        request_urn = f'{FreeNASServer.REST_API_REPLICATION}/'
+        params = {"target_dataset": target_name,
+                  "source_datasets": [source_name],"name":replication_name,
+                  "recursive": False,  "compression": None, "speed_limit":None,
+                  "enabled": True, "direction": "PUSH",  "transport": "LOCAL",  "auto":True,
```

Comment:
I wonder if auto should be False, since you manually call "run"

Reply:
Right, let me test it and fix this! thank you!

Reply:
I just tried put "auto": False, I received 422 error code from Truenas when create replication task. From the observation, the replication task will take a while (maybe minute or more) in pending state and then auto start running, I put call run api directly will save the waiting time let it run immediately. No other negative effect found so far.